### PR TITLE
[rtabmap] update to 0.20.23

### PR DESCRIPTION
**Pull request by reitowo**
Update rtabmap to 0.20.23, remove upstream merged patch

### CODE REVIEW

File: ports/rtabmap/portfile.cmake [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
 vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
 
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO introlab/rtabmap
-    REF v${VERSION}
-    SHA512 6128fa483c1ebda5a9a211f144d930ed107d0fbcc2b47c2aeace1b54f363cdabab2ef472ff20c8b2030ca067936f032714b07320488892b6424fc068dcccf280
+    # rtabmap stops releasing, check their CMakeLists.txt for version.
+    # currently is 0.20.22, this ref contains latest fix for compiling with opencv 4.7
+    REF 0e908206d04149d3bb45f27fab026a5d27e09aa8
```
Comment (JonLiu1993) [inline ports/rtabmap/portfile.cmake:8]: @cnSchwarzer, Thanks for your PR, I see that the commit corresponding to version [0.20.22](https://github.com/introlab/rtabmap/commit/a4761ad397ce195c254f65be10f7e8e2810facd0) is a4761ad397ce195c254f65be10f7e8e2810facd0, we recommend upgrading to a stable and reliable version, which is safer for users.

File: ports/rtabmap/portfile.cmake [hunk 2]
```diff
@@ -1,16 +1,15 @@
-vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}
-
 vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
 
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO introlab/rtabmap
-    REF v${VERSION}
```
Comment (JonLiu1993) [inline ports/rtabmap/portfile.cmake:8]: Could you please clarify why remove this code?

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/rtabmap/portfile.cmake (hunk 1 above)
Comment (cnSchwarzer) [reply ports/rtabmap/portfile.cmake:8]: Hi, this commit contains the patch for the removed patch `opencv47.patch`, which do you prefer? Keep the patch or use the latest commit?

File: ports/rtabmap/portfile.cmake (hunk 2 above)
Comment (cnSchwarzer) [reply ports/rtabmap/portfile.cmake:8]: Removed because I want to use the latest commit instead of the patch.

File: ports/rtabmap/vcpkg.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "rtabmap",
-  "version": "0.20.16",
+  "version": "0.20.22",
```
Comment (dg0yt) [inline ports/rtabmap/vcpkg.json:3]: There must be some suffix if this is not the official 0.20.22 source.

Comment (cnSchwarzer) [reply ports/rtabmap/vcpkg.json]: How about `0.20.22-p1`

File: ports/rtabmap/vcpkg.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "rtabmap",
-  "version": "0.20.16",
+  "version-semver": "0.20.22-p1",
```
Comment (JonLiu1993) [inline ports/rtabmap/vcpkg.json:3]: Will it report an error if we use version here?

File: ports/rtabmap/portfile.cmake (hunk 1 above)
Comment (JonLiu1993) [reply ports/rtabmap/portfile.cmake:8]: If you use commit to submit the vcpkg.json below, you cannot use the official version number 0.20.22 as the version.

File: ports/rtabmap/vcpkg.json (hunk 4 above)
Comment (cnSchwarzer) [reply ports/rtabmap/vcpkg.json]: No

File: ports/rtabmap/portfile.cmake [hunk 5]
```diff
@@ -1,16 +1,15 @@
...
-
 vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
 
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO introlab/rtabmap
-    REF v${VERSION}
-    SHA512 6128fa483c1ebda5a9a211f144d930ed107d0fbcc2b47c2aeace1b54f363cdabab2ef472ff20c8b2030ca067936f032714b07320488892b6424fc068dcccf280
+    # rtabmap stops releasing, check their CMakeLists.txt for version.
+    # currently is 0.20.22, this ref contains latest fix for compiling with opencv 4.7
```
Comment (JonLiu1993) [inline ports/rtabmap/portfile.cmake:7]:
```suggestion
```

Comment (JonLiu1993) [reply ports/rtabmap/portfile.cmake:7]: If these two lines of comments are not necessary, we can delete them.

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by JonLiu1993** [comment]
@cnSchwarzer, when I tested the features by command "./vcpkg install rtabmap[*]", I get this error.
Please take a look:
````
FAILED: bin/rtabmap-euroc_dataset.exe 
cmd.exe /C "cd . && F:\Feature-test\rtabmap\vcpkg\downloads\tools\cmake-3.25.1-windows\cmake-3.25.1-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=tools\EurocDataset\CMakeFiles\euroc_dataset.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1433~1.316\bin\Hostx64\x86\link.exe  @CMakeFiles\euroc_dataset.rsp  /out:bin\rtabmap-euroc_dataset.exe /implib:lib\rtabmap-euroc_dataset.lib /pdb:bin\rtabmap-euroc_dataset.pdb /version:0.0 /machine:X86 /nologo    /debug /INCREMENTAL /subsystem:console  && cd ."
LINK Pass 1: command "C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1433~1.316\bin\Hostx64\x86\link.exe @CMakeFiles\euroc_dataset.rsp /out:bin\rtabmap-euroc_dataset.exe /implib:lib\rtabmap-euroc_dataset.lib /pdb:bin\rtabmap-euroc_dataset.pdb /version:0.0 /machine:X86 /nologo /debug /INCREMENTAL /subsystem:console /MANIFEST /MANIFESTFILE:tools\EurocDataset\CMakeFiles\euroc_dataset.dir/intermediate.manifest tools\EurocDataset\CMakeFiles\euroc_dataset.dir/manifest.res" failed (exit code 1120) with the following output:
main.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __thiscall YAML::RepresentationException::RepresentationException(struct YAML::Mark const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (__imp_??0RepresentationException@YAML@@QAE@ABUMark@1@ABV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z) referenced in function "public: __thiscall YAML::BadSubscript::BadSubscript<char [5]>(struct YAML::Mark const &,char const (&)[5])" (??$?0$$BY04D@BadSubscript@YAML@@QAE@ABUMark@1@AAY04$$CBD@Z)
main.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __thiscall YAML::InvalidNode::InvalidNode(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (__imp_??0InvalidNode@YAML@@QAE@ABV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z) referenced in function "public: int __thiscall YAML::Node::as<int>(void)const " (??$as@H@Node@YAML@@QBEHXZ)
main.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __thiscall YAML::BadConversion::BadConversion(struct YAML::Mark const &)" (__imp_??0BadConversion@YAML@@QAE@ABUMark@1@@Z) referenced in function "public: __thiscall YAML::TypedBadConversion<int>::TypedBadConversion<int>(struct YAML::Mark const &)" (??0?$TypedBadConversion@H@YAML@@QAE@ABUMark@1@@Z)
main.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __thiscall YAML::BadConversion::BadConversion(class YAML::BadConversion const &)" (__imp_??0BadConversion@YAML@@QAE@ABV01@@Z) referenced in function "public: __thiscall YAML::TypedBadConversion<int>::TypedBadConversion<int>(class YAML::TypedBadConversion<int> const &)" (??0?$TypedBadConversion@H@YAML@@QAE@ABV01@@Z)
main.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: virtual __thiscall YAML::BadConversion::~BadConversion(void)" (__imp_??1BadConversion@YAML@@UAE@XZ) referenced in function "public: virtual __thiscall YAML::TypedBadConversion<int>::~TypedBadConversion<int>(void)" (??1?$TypedBadConversion@H@YAML@@UAE@XZ)
main.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __thiscall YAML::Node::~Node(void)" (__imp_??1Node@YAML@@QAE@XZ) referenced in function _main
main.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: struct YAML::Mark __thiscall YAML::Node::Mark(void)const " (__imp_?Mark@Node@YAML@@QBE?AU02@XZ) referenced in function "public: int __thiscall YAML::as_if<int,void>::operator()(void)const " (??R?$as_if@HX@YAML@@QBEHXZ)
main.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: enum YAML::NodeType::value __thiscall YAML::Node::Type(void)const " (__imp_?Type@Node@YAML@@QBE?AW4value@NodeType@2@XZ) referenced in function "public: static bool __cdecl YAML::convert<int>::decode(class YAML::Node const &,int &)" (?decode@?$convert@H@YAML@@SA_NABVNode@2@AAH@Z)
main.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: bool __thiscall YAML::Node::IsNull(void)const " (__imp_?IsNull@Node@YAML@@QBE_NXZ) referenced in function _main
main.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: bool __thiscall YAML::Node::IsScalar(void)const " (__imp_?IsScalar@Node@YAML@@QBE_NXZ) referenced in function "public: static bool __cdecl YAML::convert<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > >::decode(class YAML::Node const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &)" (?decode@?$convert@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@YAML@@SA_NABVNode@2@AAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z)
main.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const & __thiscall YAML::Node::Scalar(void)const " (__imp_?Scalar@Node@YAML@@QBEABV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@XZ) referenced in function "public: static bool __cdecl YAML::convert<int>::decode(class YAML::Node const &,int &)" (?decode@?$convert@H@YAML@@SA_NABVNode@2@AAH@Z)
main.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: unsigned int __thiscall YAML::Node::size(void)const " (__imp_?size@Node@YAML@@QBEIXZ) referenced in function _main
  Hint on symbols that are defined and could potentially match:
````

**Comment by cnSchwarzer** [comment]
@JonLiu1993 Hi, fixed!

**Comment by cnSchwarzer** [comment]
Fixed

**Comment by cnSchwarzer** [comment]
FYI: This ref is official 0.20.23

**Comment by cnSchwarzer** [comment]
@BillyONeal Ready for review, the former reviewer seems on holiday?

**Comment by BillyONeal** [comment]
rtabmap[core]:x64-windows pass
rtabmap[core,tools]:x64-windows pass
rtabmap[core,opencv-contrib]:x64-windows pass
rtabmap[core,tools,opencv-contrib]:x64-windows pass

Thanks for the update!

**Comment by cnSchwarzer** [comment]
Thanks for merging!

**Pull request merged**